x/programs/rust/examples/single_nft/scripts/build_and_test.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+if ! [[ "$0" =~ scripts/build_and_test.sh ]]; then
+  echo "must be run from crate root"
+  exit 255
+fi
+
+root="$(pwd)"
+
+simulator_path="${PWD}"/../../../cmd/simulator
+simulator_bin="${simulator_path}"/bin/simulator
+
+echo "Downloading dependencies..."
+cd "${simulator_path}"
+go mod download
+
+echo "Building Simulator..."
+go build -o "${simulator_bin}" "${simulator_path}"/simulator.go
+
```

Comment:
Without going back to the root, you'll try to run the token tests as well. 

```suggestion

cd $root
```

Reply:
I fixed this by specifying the single_nft crate